### PR TITLE
Where op

### PR DESCRIPTION
**Pull request by scxiao**
Used on distillbert and gpt-2-ml model

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -2424,11 +2426,22 @@ struct onnx_parser
...
         return l;
     }
 
+    instruction_ref
+    parse_where(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        auto type = args[1]->get_shape().type();
+        auto cond = prog.add_instruction(
+            make_op("convert", {{"target_type", shape::type_string(type)}}), args[0]);
+        auto diff = add_broadcastable_binary_op(args[1], args[2], "sub");
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:2440]: can we make a comment in the code of the equation being used that is equivalent to the conditional statement?

Comment (scxiao) [reply src/onnx/onnx.cpp:2442]: add one

File: src/include/migraphx/serialize.hpp [hunk 2]
```diff
@@ -102,8 +103,15 @@ auto to_value_impl(rank<10>, const T& x) -> decltype(x.to_value())
     return x.to_value();
 }
 
+template <class T, MIGRAPHX_REQUIRES(std::is_same<T, shape::type_t>{})>
+value to_value_impl(rank<11>, const T& x)
```
Comment (pfultz2) [inline src/include/migraphx/serialize.hpp:107]: Why is this overload added? The enum overload [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/serialize.hpp#L88) should handle this.

File: src/include/migraphx/serialize.hpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include <migraphx/reflect.hpp>
 #include <migraphx/requires.hpp>
 #include <migraphx/rank.hpp>
+#include <migraphx/shape.hpp>
```
Comment (pfultz2) [inline src/include/migraphx/serialize.hpp:9]: This header shouldn't depend on the shape class.

File: src/include/migraphx/serialize.hpp (hunk 2 above)
Comment (scxiao) [reply src/include/migraphx/serialize.hpp:107]: Without the overload, we need a constructor for the value with an enum as the input...

**Review by pfultz2 — APPROVED** [review]

**Review by kahmed10 — APPROVED** [review]

**Pull request merged**